pkg/api/steve/catalog/plugin_test.go

```diff
@@ -0,0 +1,40 @@
+package catalog
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestProxyRequest_content_type(t *testing.T) {
+	response := "var http = require('http');\n        var url = require('url');\n        var number = 0;"
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Disposition", "attachment; filename=testing.js")
+		if _, err := w.Write([]byte(response)); err != nil {
+			t.Fatal(err)
+		}
+	}))
+	defer ts.Close()
+	denyFunc = func(string) bool { return false }
```

Comment:
You need to store the old one, and put it back, otherwise any additional tests would have undefined behaviour.

This is why passing in the verifier is a better option .

Reply:
That's true, I updated the code